metricbeat/module/redis/keyspace/keyspace.go

```diff
@@ -42,43 +37,27 @@ func init() {
 
 // MetricSet for fetching Redis server information and statistics.
 type MetricSet struct {
-	mb.BaseMetricSet
-	pool *rd.Pool
+	*redis.MetricSet
 }
 
 // New creates new instance of MetricSet
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	// Unpack additional configuration options.
-	config := struct {
-		IdleTimeout time.Duration `config:"idle_timeout"`
-		Network     string        `config:"network"`
-		MaxConn     int           `config:"maxconn" validate:"min=1"`
-		Password    string        `config:"password"`
-	}{
-		Network:  "tcp",
-		MaxConn:  10,
-		Password: "",
-	}
-	err := base.Module().UnpackConfig(&config)
+	ms, err := redis.NewMetricSet(base)
 	if err != nil {
 		return nil, err
 	}
-
-	return &MetricSet{
-		BaseMetricSet: base,
-		pool: redis.CreatePool(base.Host(), config.Password, config.Network,
-			config.MaxConn, config.IdleTimeout, base.Module().Config().Timeout),
-	}, nil
+	return &MetricSet{ms}, nil
 }
 
 // Fetch fetches metrics from Redis by issuing the INFO command.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+func (m *MetricSet) Fetch(r mb.ReporterV2) {
 	// Fetch default INFO.
-	info, err := redis.FetchRedisInfo("keyspace", m.pool.Get())
+	info, err := redis.FetchRedisInfo("keyspace", m.Connection())
 	if err != nil {
-		return nil, err
+		logp.Err("Failed to fetch redis info: %s", err)
+		return
```

Comment:
No reporting of the error? Should we include something in the error message that it was a request for `keyspace`?

Reply:
Agree with @ruflin that as much context of an error must be included. This error only happens when requesting _keyspace_ and we can't control error messages from Redis itself. It is potentially a non descriptive error for our users so I'd add more info here to help debugging and report the error too.

Reply:
Added mention to keyspaces.